BlueprintUI/Sources/Environment/Environment.swift

```diff
@@ -73,6 +151,51 @@ public struct Environment {
     }
 }
 
+extension Environment {
+
+    struct Subset {
+        var values: [StorageKey: ValueBox] = [:]
+    }
+
+    /// Place values in the environment into a reference box, so that their storage is only allocated once
+    /// even when placed into an environment subset for change tracking.
+    final class ValueBox {
+        let base: Any
+
+        init(_ base: Any) {
+            self.base = base
+        }
+    }
+
+    struct StorageKey: Hashable {
+
+        private let identifier: ObjectIdentifier
+
+        private let isEqual: (Any?, Any?) -> Bool
+        private let keyTypeName: () -> String
```

Comment:
This is my bad (sorry), but I think we should just pass the type in here instead, probably.